idaes/core/util/model_diagnostics.py

```diff
@@ -251,6 +266,14 @@ def svd_sparse(jacobian, number_singular_values):
         description="Tolerance for raising a warning for small Jacobian values.",
     ),
 )
+CONFIG.declare(
+    "strict_evaluation_error_detection",
+    ConfigValue(
+        default=True,
+        domain=bool,
+        description="If False, warnings will not be generated for things like log(x) with x >= 0",
+    ),
+)
```

Comment:
I'm not sure what the default should be here. For Ipopt, with default options (i.e. `bound_relax_factor=1e-8`), we are strictly feasible path with respect to bounds of zero, but not other bounds (at least according to this option's documentation). So something like `log(1 - x); x <= 1` could still hit the evaluation error. I think I lean towards treating this as an advanced option that people can turn on if they know what they're doing, although I think we will have to test this in the wild to really know what the "right" default is.

Reply:
I tested the example you mentioned with x initialized to 2, and I did not get an evaluation error (if I set the nlp_scaling_method to none).

Reply:
Which is consistent with the way we do fraction to the boundary in parapint.

Reply:
That said, I agree with your overall point.

Reply:
> I tested the example you mentioned with x initialized to 2, and I did not get an evaluation error (if I set the nlp_scaling_method to none).

What if you set `bound_push` to zero (or anything less than `1*bound_relax_factor`, I think)?

The point is just that the evaluation error *could* happen.

Reply:
I think we are agreeing? The default will produce a warning for the example you gave.

Reply:
Oh I interpreted the name of the option as "treat bounds as if they are strict". But instead, it is "be more strict about the region where evaluation errors occur"?

Reply:
Exactly. Naming...

Reply:
I would prefer the name to describe how the bounds are treated, rather than how the evaluation error region is treated (as this is mathematically constant). What about `"eval_error_strict_bounds"`? I'm also open to other arguments/proposals.

Reply:
Just to be clear, are you suggesting renaming and reversing the meaning?

Reply:
Yes, I'm suggesting renaming the option, although I'd prefer to get another opinion, as naming preferences can be fairly idiosyncratic.

I would be fine with a name that describes the "region of evaluation error", but would prefer to call it something other than "strict" as this makes me think of how the bounds are handled (unless others think the name makes sense).

Reply:
Personally I read "strict_evaluation_error_detection" in the sense that @michaelbynum used it, but equally when I read the doc string I asked "what exactly happens when it is set to `False` and why?" This does have me leaning towards @Robbybp's suggestion as would focus the documentation on "why you might want to set this to relax the definition of bounds".

Reply:
Since this is a potentially niche use case, maybe we should defer including this option for now (and keep the default behavior of treating bounds as if they are non-strict), then add this later if we find that the output is often dominated by bounds of zero that we don't actually expect to hit.

Reply:
@andrewlee94, are you okay with removing the option for now and adding later?

Reply:
That is fine by me.